.ci/scripts/util.sh

```diff
@@ -2,65 +2,129 @@
-    # echo "SECONDS,disk_used,swap_used,total_threads,last_line" > $metrics_misc
 
     # start the background processes to monitor docker stats
-    # TODO For the time being (may ultimately remove) take out the sleep ("$(metrics_seconds),$ds" >> ${metrics_stats}; sleep .1; done })
-    { while true; do ds=$(docker stats --no-stream --format "${stat_format}" "${container_name}" 2>/dev/null); echo "$(metrics_seconds)","$ds" >> "${metrics_stats}"; done } &
-    echo "$!" > "${container_name}_metrics_stats_bg_pid.txt"
+    { while true; do ds=$(docker stats --no-stream --format "${stat_format}" "${pge}" 2>/dev/null); \
+    echo "$(metrics_seconds)","$ds" >> "${metrics_stats}"; sleep "$sample_time"; done } & \
+    echo "$!" > "${container_info}_metrics_stats_bg_pid.txt"
+
+    # Miscellaneous Statistics
+
+    # test for MAC operating system for OS statistics
+    if [[ $OSTYPE == "darwin"* ]]
+    then
+        echo "Mac Operating system: $OSTYPE"
+        echo "SECONDS,disk_used,total_threads,last_line" > "$metrics_misc"
+
+        block_space_cmd='df -B 1024 | grep "/System/Volumes/VM"'
+
+        sys_threads_cmd='ps -elf | wc -l'
+
+        # Set directory fo the log file
+        last_log_line_dir='pwd'
+
+        lll_file="last_log_line.txt"
+        lll_cmd="echo $(find ${last_log_line_dir} -name ${lll_file} -exec rm {} \; 2>/dev/null)"
+
+        { while true; do dus=$(eval "$block_space_cmd"); ths=$(eval "$sys_threads_cmd"); lll=$(eval "$lll_cmd"); \
+        echo "$(metrics_seconds),$dus,$ths,$lll" >> "${metrics_misc}"; sleep "$sample_time"; done } & \
+        echo "$!" > "${container_info}_metrics_misc_bg_pid.txt"
+    else
+        echo "Linux Operating system: $OSTYPE"
+        echo "SECONDS, disk_used, swap_used, total_threads, last_line" > "$metrics_misc"
+        # Use 'df' command to capture the amount of space on the '/dev/vda1' file system (-B sets block size (1K)
+        # the line represent Filesystem  1K-blocks  Used Blocks Available Blocks %Used Mounted_On
+        block_space_cmd='df -B 1024 | grep "/dev/vda1"'
+
+        # Get the number of system threads
+        sys_threads_cmd='ps -elf | wc -l'
+
+        # Use 'free' to get the total amount of Swap space available
+        swap_space_cmd='free -g | grep Swap'
+
+        # Set directory fo the log file
+        last_log_line_dir='pwd'
+
+        lll_file="last_log_line.txt"
+        find ${last_log_line_dir} -name ${lll_file} -exec rm {} \; 2>/dev/null
+        lll_cmd="echo $(find ${last_log_line_dir} -name ${lll_file} -exec rm {} \; 2>/dev/null)"
+
+        { while true; do dus=$(eval "$block_space_cmd"); swu=$(eval "$swap_space_cmd"); ths=$(eval "$sys_threads_cmd"); \
+        lll=$(eval "$lll_cmd"); echo "$(metrics_seconds), $dus, $swu, $ths, $lll" >> "${metrics_misc}"; \
+        sleep "$sample_time"$; done } & \
+        echo "$!" >> "${container_info}_metrics_misc_bg_pid.txt"
+    fi
+
 }
 
-metrics_collection_end ()
+metrics_collection_end()
```

Comment:
Would be good to have a comment that describes the input arguments, similar to what is in `metrics_collection_start()`

Reply:
Done